src/averageAges.js

```diff
@@ -55,7 +67,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filterDifferents = !onlyWithSon
+    ? people.filter(({ mother }) => (
+      people.find(({ name }) => mother === name)))
+    : people.filter(({ mother, sex }) => (
+      people.find(({ name }) => mother === name)
+      && sex === 'm'));
```

Comment:
```suggestion
  const filterDifferents = people.filter(({ mother, sex }) => {
    return people.some(({ name }) => mother === name)
      && (onlyWithSon ? sex === 'm' : true);
  });
```

This can be optimized too.